source/assets/js/srf-chmap.js

```diff
@@ -36,51 +36,41 @@ var chmapController = function() {
 
     this.initObservers = function() {
 
-        // tooltips for cantons
-        $('.chmap-wrapper .chmap--desktop').on('mousemove', function (event) {
-            var $target = $(event.target);
-            if ($target.hasClass("chmap--desktop")) {
-                return
-            }
+        $('.chmap__location', '.chmap--desktop').on('mousemove', function (event) {
+
             $(':focus').focusout();
-            var parentOffset = $('.chmap--desktop').parent().offset(),
-            	pageY = event.pageY - parentOffset.top - 58,
-                pageX = event.pageX - parentOffset.left,
-                $tooltip = $("#chmap-tooltip"),
-                mapId = that.getCurrentMapId($target),
-                map = that.getMapById(mapId),
-                canton = null;
-            if ($target.is("use")) {
-                canton = map.getCantonById(that.extractCantonId($target.attr("xlink:href").substr(1)));
-            } else {
-                canton = map.getCantonById(that.extractCantonId($target.parent().attr("id")));
-            }
 
-            // if not there yet: create it
-            if ($tooltip.length === 0) {
-                // TODO: check what happens if there are two maps!
-                $("#map-" + mapId).append('<div id="chmap-tooltip" class="chmap-tooltip"></div>');
-                $tooltip = $("#chmap-tooltip"); // :(
-            }
+            var $wrapper = $(this).closest('.chmap-wrapper'),
+                $tooltip = $('.js-chmap-tooltip', $wrapper),
+                parentOffset = $wrapper.offset(),
+                pageY = event.pageY - parentOffset.top - 58,
+                pageX = event.pageX - parentOffset.left,
+                cssClass = "left";
 
-            // tooltip positioning
-            var cssClass = "left";
-            if (pageX > $(this).parent().width() / 2) {
+            // position of the tooltip
+            if (pageX > $wrapper.width() / 2) {
                 cssClass = "right";
                 pageX = pageX - $tooltip.outerWidth();
             }
             $tooltip.css({"top": pageY, "left": pageX})
                 .removeClass("chmap-tooltip--left chmap-tooltip--right")
                 .addClass("chmap-tooltip--" + cssClass);
 
-            // naming it
-            $tooltip.html(canton.name);
-            $tooltip.show();
+            // get canton name and show tooltip - only if the map is available
+            var cantonId = $(this).attr('id').split("-")[0];
```

Comment:
FeF kann doch auch ES6, oder ? (var und that sollten daher nicht notwendig sein)